### PR TITLE
OCPCLOUD-2600: Bump deps to k8s v1.30, go 1.22

### PR DESCRIPTION
**Pull request by damdo**
Bump deps to k8s v1.30, go 1.22

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,9 +1,9 @@
-FROM registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.21-openshift-4.16 AS builder
+FROM registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.22-openshift-4.16 AS builder
```
Comment (JoelSpeed) [inline Dockerfile:1]:
```suggestion
FROM registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.22-openshift-4.17 AS builder
```

**Review by JoelSpeed — COMMENTED** [review]
Looks good but I think you need to update the image to 4.17

File: Dockerfile (hunk 1 above)
Comment (damdo) [reply Dockerfile]: Yeah, missed it, TY.

File: .golangci.yaml [hunk 2]
```diff
@@ -38,7 +38,6 @@ linters:
     - goimports
     - goprintffuncname
     - gosec
-    - ifshort
```
Comment (JoelSpeed) [inline .golangci.yaml:41]: Was this deprecated?

Comment (damdo) [reply .golangci.yaml:41]: Yes

**Comment by JoelSpeed** [comment]
Assuming tests start going green, LGTM

**Pull request merged**